addon/mixins/relationship-support-model.js

```diff
@@ -0,0 +1,50 @@
+import getRelationshipDescriptor from '@bagaar/ember-data-bagaaravel/utils/get-relationship-descriptor';
+import { assert } from '@ember/debug';
+import Mixin from '@ember/object/mixin';
+
+export default Mixin.create({
+  /**
+   * Methods
+   */
+
+  saveRelationship(relationshipName) {
+    assert(
+      'ember-data-bagaaravel: Cannot save a relationship of a newly created record.',
+      !this.isNew,
+    );
+
+    assert(
+      `ember-data-bagaaravel: "${relationshipName}" is not a valid relationship name.`,
+      getRelationshipDescriptor(this, relationshipName),
+    );
+
+    assert(
+      `ember-data-bagaaravel: "${relationshipName}" relationship can not be serialized.`,
+      this.canSerializeRelationship(relationshipName),
+    );
+
+    return this.save({
+      adapterOptions: {
+        relationshipName,
+      },
+    });
+  },
+
+  saveRelationships(...relationshipNames) {
+    assert(
+      'ember-data-bagaaravel: Cannot save relationships of a newly created record.',
+      !this.isNew,
+    );
+
+    let promises = relationshipNames.map(relationshipName => this.saveRelationship(relationshipName));
+
+    return Promise.all(promises).then(() => this);
```

Comment:
Decided to use native promises based on: https://caniuse.com/#feat=promises.

Reply:
Ah, ok. I was thinking about the runloop stuff, but I don't know enough about that. I assume it's not a problem since we also use async await in projects.